### PR TITLE
feat(tree): add switcherLoadingIcon prop

### PR DESCRIPTION
**Pull request by coding-ice**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/48681

### 📝 Changelog

- Use a developer-oriented tone and narrative style.
- Describe the user's first-hand experience of the issue and its impact on developers, rather than your solution approach.
- Refer to: ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | support custom loading icon           |
| 🇨🇳 Chinese | 支持自定义loading icon           |

### CODE REVIEW

**Comment by afc163** [comment]
图片

**Comment by afc163** [comment]
Conflicting

File: components/tree/index.en-US.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | showIcon | Controls whether to display the `icon` node, no default style | boolean | false |  |
 | showLine | Shows a connecting line | boolean \| {showLeafIcon: ReactNode \| ((props: AntTreeNodeProps) => ReactNode)} | false |  |
 | switcherIcon | Customize expand/collapse icons for tree nodes (With default rotate angular style) | ReactNode \| ((props: AntTreeNodeProps) => ReactNode) | - | renderProps: 4.20.0 |
+| switcherLoadingIcon | Customize loading icons for tree nodes | ReactNode | - |  |
```
Comment (li-jia-nan) [inline components/tree/index.en-US.md:70]:
```suggestion
| switcherLoadingIcon | Customize loading icons for tree nodes | ReactNode | - | 5.20.0 |
```

File: components/tree/utils/iconUtil.tsx [hunk 2]
```diff
@@ -13,15 +13,19 @@ interface SwitcherIconProps {
...
 }
 
 const SwitcherIconCom: React.FC<SwitcherIconProps> = (props) => {
-  const { prefixCls, switcherIcon, treeNodeProps, showLine } = props;
+  const { prefixCls, switcherIcon, treeNodeProps, showLine, switcherLoadingIcon } = props;
 
   const { isLeaf, expanded, loading } = treeNodeProps;
 
   if (loading) {
+    if (React.isValidElement(switcherLoadingIcon)) {
```
Comment (li-jia-nan) [inline components/tree/utils/iconUtil.tsx:26]: 这个改成三元表达式么？

Comment (coding-ice) [reply components/tree/utils/iconUtil.tsx:26]: 之前是三元，被`eslint:no-unneeded-ternary`拦掉了

Comment (li-jia-nan) [reply components/tree/utils/iconUtil.tsx:26]: 那就不管了

**Review by afc163 — APPROVED** [review]

**Pull request merged**